### PR TITLE
FIX: Disallow trash abilities on accessing agendas in archives

### PR DESCRIPTION
**Pull request by NoahTheDuke**
Missed this when building the Access prompts.

### CODE REVIEW

File: test/clj/game_test/cards/programs.clj [hunk 1]
```diff
@@ -379,7 +379,16 @@
...
+        (is (= 0 (:agenda-point (get-runner))) "Runner did not steal TFP"))))
+  (testing "vs cards in Archives"
+    (do-game
+      (new-game (default-corp ["Hostile Takeover"])
+                (default-runner ["Imp"]))
+      (core/move state :corp (find-card "Hostile Takeover" (:hand (get-corp))) :discard)
+      (take-credits state :corp)
+      (play-from-hand state :runner "Imp")
+      (run-empty-server state "Archives")
+      (is (= ["Steal"] (->> (get-runner) :prompt first :choices)) "Should only get the option to steal"))))
```
Comment (Saintis) [inline test/clj/game_test/cards/programs.clj]: I'd add "when accessing Hostile Takeover from archives"

File: src/clj/game/core/runs.clj [hunk 2]
```diff
@@ -218,9 +218,10 @@
         cost-as-symbol (when (= 1 (count cost-strs)) (costs-to-symbol cost))
         ;; any trash abilities
         can-steal-this? (can-steal? state side c)
-        trash-ab-cards (->> (concat (all-active state :runner)
-                                    (get-in @state [:runner :play-area]))
-                            (filter #(can-trigger? state :runner (:trash-ability (:interactions (card-def %))) % [c])))
+        trash-ab-cards (when (not= (:zone c) [:discard])
+                         (->> (concat (all-active state :runner)
+                                      (get-in @state [:runner :play-area]))
+                              (filter #(can-trigger? state :runner (:trash-ability (:interactions (card-def %))) % [c]))))
```
Comment (Saintis) [inline src/clj/game/core/runs.clj]: I'd change the nested keywords to a `get-in`.

File: test/clj/game_test/cards/programs.clj (hunk 1 above)
Comment (NoahTheDuke) [reply test/clj/game_test/cards/programs.clj]: Why?

Comment (Saintis) [reply test/clj/game_test/cards/programs.clj]: More explicit, and easier to understand what the issue is when the test fails

File: src/clj/game/core/runs.clj [hunk 3]
```diff
@@ -221,7 +221,7 @@
         trash-ab-cards (when (not= (:zone c) [:discard])
                          (->> (concat (all-active state :runner)
                                       (get-in @state [:runner :play-area]))
-                              (filter #(can-trigger? state :runner (:trash-ability (:interactions (card-def %))) % [c]))))
+                              (filter #(can-trigger? state :runner (->> % card-def :interactions :trash-ability) % [c]))))
```
Comment (Saintis) [inline src/clj/game/core/runs.clj]: This double arrow `->>` could be a single arrow `->` to save a valuable character

File: src/clj/game/core/runs.clj (hunk 2 above)
Comment (NoahTheDuke) [reply src/clj/game/core/runs.clj]: Heh yeah, I’ll probably go back to `get-in` next time I’m on the computer. I realized I could pull one more level out without thinking about how it’d read later. I personally like how it looks, but only when it’s... conversational? Like when checking prompts in a test. This isn’t quite that, so I think you’re right.

File: src/clj/game/core/runs.clj (hunk 3 above)
Comment (NoahTheDuke) [reply src/clj/game/core/runs.clj]: Honestly, I just forgot that the single threading macro places it immediately after the func. I keep thinking it places it to the left of the function, lol.

File: test/clj/game_test/cards/programs.clj (hunk 1 above)
Comment (NoahTheDuke) [reply test/clj/game_test/cards/programs.clj]: Makes sense! I’ll pay attention to that in the future.

**Pull request merged**